source/jormungandr/jormungandr/street_network/tests/valhalla_test.py

```diff
@@ -0,0 +1,343 @@
+                        url='http://bob.com',
+                        costing_options={'bib': 'bom'})
+    resp_json = response_valid()
+    origin = get_pt_object(type_pb2.ADDRESS, 2.439938, 48.572841)
+    destination = get_pt_object(type_pb2.ADDRESS, 2.440548, 48.57307)
+    response = valhalla._get_response(resp_json, 'walking',
+                                      origin,
+                                      destination,
+                                      str_to_time_stamp('20161010T152000'))
+    assert response.status_code == 200
+    assert response.response_type == response_pb2.ITINERARY_FOUND
+    assert len(response.journeys) == 1
+    assert response.journeys[0].duration == 6
+    assert len(response.journeys[0].sections) == 1
+    assert response.journeys[0].sections[0].type == response_pb2.STREET_NETWORK
+    assert response.journeys[0].sections[0].type == response_pb2.STREET_NETWORK
+    assert response.journeys[0].sections[0].length == 52
+    assert response.journeys[0].sections[0].destination == destination
+
+
+def direct_path_func_without_response_valhalla_test():
+    instance = MagicMock()
+    valhalla = Valhalla(instance=instance,
+                        url='http://bob.com',
+                        costing_options={'bib': 'bom'})
+    valhalla.breaker = MagicMock()
+    valhalla._call_valhalla = MagicMock(return_value=None)
+    valhalla._format_url = MagicMock(return_value=valhalla.service_url)
+    with pytest.raises(TechnicalError) as excinfo:
+        valhalla.direct_path(None, None, None, None, None)
+    assert '500: Internal Server Error' == str(excinfo.value)
+    assert 'TechnicalError' == str(excinfo.typename)
+
+
+def direct_path_func_with_status_code_400_response_valhalla_test():
+    instance = MagicMock()
+    valhalla = Valhalla(instance=instance,
+                        url='http://bob.com',
+                        costing_options={'bib': 'bom'})
+    valhalla.breaker = MagicMock()
+    response = requests.Response()
+    response.status_code = 400
+    response.url = valhalla.service_url
+    valhalla._call_valhalla = MagicMock(return_value=response)
+    valhalla._format_url = MagicMock(return_value=valhalla.service_url)
+
+    nav_resp = valhalla.direct_path(None, None, None, None, None)
+
+    assert response.status_code == nav_resp.status_code
+    assert nav_resp.error.message == 'Valhalla service unavailable, impossible to query : http://bob.com'
+
+
+def direct_path_func_with_valid_response_valhalla_test():
+    instance = MagicMock()
+    valhalla = Valhalla(instance=instance,
+                        url='http://bob.com',
+                        costing_options={'bib': 'bom'})
+    resp_json = response_valid()
+
+    valhalla.breaker = MagicMock()
+    response = requests.Response()
+    response.status_code = 200
+    response.json = resp_json
+    response.url = valhalla.service_url
+    valhalla._call_valhalla = MagicMock(return_value=response)
```

Comment:
I would prefer that you use requests-mock, this way we also test the call to requests, it will ensure that we don't mess in the function ```_call_valhalla```, you can found a example here: https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/jormungandr/parking_space_availability/bss/tests/jcdecaux_test.py#L100-L103